src/libraries/Common/src/System/Text/Json/PooledByteBufferWriter.cs

```diff
@@ -22,6 +22,12 @@ internal sealed class PooledByteBufferWriter : IBufferWriter<byte>, IDisposable
 
         private const int MinimumBufferSize = 256;
 
+#if NETCOREAPP
+        private static readonly int MaximumBufferSize = Array.MaxLength;
```

Comment:
It would be good to have an outer loop test to verify these 2 limits, at least a test to make sure we can allocate up to that point with or without JSON.

Reply:
I've simplified to just one constant. The added test confirms we can't exceed the limit. It also tries to test just below the limit but accounts for the test machine not being able to allocate that amount.